lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_StatePMPSEnables.TcPOU

```diff
@@ -129,9 +139,17 @@ ffNoGoal(
     This action sets bForwardEnable and bBackwardEnable based on
     the current position and the calculated bounds.
 *)
-IF bValidGoal AND bEnable THEN
-    bForwardEnabled := stMotionStage.stAxisStatus.fActPosition < fUpperPos;
-    bBackwardEnabled := stMotionStage.stAxisStatus.fActPosition > fLowerPos;
+IF bLockBounds THEN
+    // Prevent forward/backward motion if the position of the axis is outside of the upper/lower bounds respectively.
+    // Prevent forward/backward motion if the command to the axis is outside of the upper/lower bounds respectively.
+    bForwardEnabled := stMotionStage.stAxisStatus.fActPosition < fUpperPos AND stMotionStage.fPosition < fUpperPos;
+    bBackwardEnabled := stMotionStage.stAxisStatus.fActPosition > fLowerPos AND stMotionStage.fPosition > fLowerPos;
```

Comment:
I'm waffling a bit in my head if it might be better to use the direct sources for these numbers rather than the EPICS status and setpoint fields. In practice these are essentially always the same but I have to wonder if there are any edge cases where either this or the raw values as reported by the NC are more correct to use.

Maybe this is best as-is due to interface flexibility (technically not requiring the NC)

But the other option would have been to use `stMotionStage.Axis.NcToPlc.ActPos` and `stMotionStage.Axis.NcToPlc.TargetPos`

Food for thought, almost didn't even post this

Edit: another thing to consider is that fPosition here is not valid until bExecute goes to TRUE, which feels like a rare edge case that could occur if someone changed code elsewhere.

Reply:
I switched it to use the NC values directly. I think that makes sense